### PR TITLE
Docs: Add some docstrings for ADC module.

### PR DESCRIPTION
**Pull request by phuongtran7**
From #446, add some docstrings for the ADC module. Mostly just add the example and some small explanation.

### CODE REVIEW

File: src/adc.rs [hunk 1]
```diff
@@ -456,6 +458,59 @@ impl_adc!(ADC1: adc_unit_t_ADC_UNIT_1);
...
+///     let mut adc_pin = AdcChannelDriver::new(&adc, peripherals.pins.gpio12, &config)?;
+
+///     loop {
+///         /// you can change the sleep duration depending on how often you want to sample
+///         thread::sleep(Duration::from_millis(100));
+///         println!("ADC value: {}", adc.read(&mut adc_pin)?);
+///     }
+/// }
+
+/// #[cfg(any(feature = "adc-oneshot-legacy", esp_idf_version_major = "4"))]
```
Comment (Vollbrecht) [inline src/adc.rs:503]: This part does not need to be included-

**Review by Vollbrecht — CHANGES_REQUESTED** [review]
Thanks for the PR. Please remove the part where we have two mains. as i think for the docstring we should only show what is needed and not what cant work with that  particular config.

File: src/adc.rs (hunk 1 above)
Comment (phuongtran7) [reply src/adc.rs]: Thank you so much for the quick look. The section should be removed now.

File: src/adc.rs [hunk 2]
```diff
@@ -456,6 +458,48 @@ impl_adc!(ADC1: adc_unit_t_ADC_UNIT_1);
 #[cfg(not(any(esp32c2, esp32h2, esp32c5, esp32c6, esp32p4)))] // TODO: Check for esp32c5 and esp32p4
 impl_adc!(ADC2: adc_unit_t_ADC_UNIT_2);
 
+/// One-shot ADC module
+/// Example: reading a value form a pin and printing it on the terminal
+/// ```
+/// use std::thread;
+/// use std::time::Duration;
+
+/// #[cfg(not(any(feature = "adc-oneshot-legacy", esp_idf_version_major = "4")))]
```
Comment (Vollbrecht) [inline src/adc.rs:467]: should probabbly also remove this cfg(not(any .. as a user would only look at this if the module itself is visible in the first place,

**Review by Vollbrecht — COMMENTED** [review]
Besides the one small nit i think it looks good.

File: src/adc.rs (hunk 2 above)
Comment (phuongtran7) [reply src/adc.rs]: Opps, sorry. Should have looked closer. All `cfg` lines should be removed now. And I did a quick check to make the examples look somewhat similar too.

**Review by Vollbrecht — APPROVED** [review]

**Comment by Vollbrecht** [comment]
Thanks!

**Pull request merged**